include/functional/inspect.hpp

```diff
@@ -34,18 +35,16 @@ struct inspect_t::apply final {
                                decltype(std::as_const(v.error()))>
              && (!std::is_void_v<decltype(v.value())>)
   {
-    static_assert(
-        std::is_void_v<std::invoke_result_t<
-            decltype(fn1), decltype(std::forward<decltype(v)>(v).value())>>);
-    static_assert(
-        std::is_void_v<std::invoke_result_t<
-            decltype(fn2), decltype(std::forward<decltype(v)>(v).error())>>);
+    static_assert(std::is_void_v<std::invoke_result_t<
+                      decltype(fn1), decltype(std::as_const(v.value()))>>);
+    static_assert(std::is_void_v<std::invoke_result_t<
+                      decltype(fn2), decltype(std::as_const(v.error()))>>);
     if (v.has_value()) {
-      std::forward<decltype(fn1)>(fn1)(std::as_const(v.value()));
+      FWD(fn1)(std::as_const(v.value()));
```

Comment:
why the as_const? Moving from temporary inspects is kinda fundamental, no?

Reply:
This could be perhaps rewritten as `FWD(fn1)(std::as_const(v).value());` ; the point of `inspect` is that it is meant for side-effects only, no mutations allowed. Perhaps it needs sibling operations `inspect_value` and `inspect_error`, same purpose but shortcircuit as needed.

Perhaps you confused it with another primitive which I am yet to implement, and which you might know as `match` or `visit`. It will be a sink for the input monad and produce a new value (which might not even be a monad, or more typically would be a monad of a different kind)

Reply:
come to think of it, perhaps we do not need `inspect` as implemented here at all, and want it to shortcircuit instead (as I suggested above `inspect_value` would do).